### PR TITLE
feat: newpair, rename commands for crosspost

### PR DESCRIPTION
**Pull request by XangelMusic**
Added modifications and methods to accommodate the new crosspost pair feature. Included documentation for most group commands.

### CODE REVIEW

File: commands/user.go [hunk 1]
```diff
@@ -131,283 +164,445 @@ func userGroup(b *bot.Bot) {
...
+			if err != nil {
+				return messages.ErrChannelNotFound(err, child)
+			}
+
+			// Checks if discord channel is a text channel.
+			if ch.Type != discordgo.ChannelTypeGuildText {
+				return messages.ErrIncorrectCmd(ctx.Command)
+			}
+
+			// Checks if crosspost channel is not parent channel.
```
Comment (VTGare) [inline commands/user.go:276]: Using this comment as an example, but there's a lot of more of them.

These comments are excessive and don't really explain anything, in this particular case `if parent == child` is already obvious what it does just by looking at it.  Some comments can be left, but I'll have to go through them myself.

File: commands/user.go [hunk 2]
```diff
@@ -131,283 +164,445 @@ func userGroup(b *bot.Bot) {
...
+			children = append(children, child)
+		}
+
+		// Creates crosspost group when checks suffice.
 		_, err = b.Store.CreateCrosspostGroup(context.Background(), user.ID, &store.Group{
 			Name:     name,
 			Parent:   parent,
-			Children: []string{},
+			Children: children,
+			IsPair:   ispair,
```
Comment (VTGare) [inline commands/user.go:295]: Rename local ispair to isPair

File: commands/user.go [hunk 3]
```diff
@@ -131,283 +164,445 @@ func userGroup(b *bot.Bot) {
...
 
+// delgroup deletes a crosspost group.
 func delgroup(b *bot.Bot) func(ctx *gumi.Ctx) error {
 	return func(ctx *gumi.Ctx) error {
+		// Checks if Boe Tea command is correct.
 		if ctx.Args.Len() < 1 {
 			return messages.ErrIncorrectCmd(ctx.Command)
 		}
 
+		// Assigns current discord user.
```
Comment (VTGare) [inline commands/user.go:326]: Some comments are also just misleading, the user isn't assigned it's retrieved from the database.

File: commands/user.go [hunk 4]
```diff
@@ -131,283 +164,445 @@ func userGroup(b *bot.Bot) {
...
+		// Prepares new embed builder to list crosspost groups.
 		locale := messages.UserGroupsEmbed(ctx.Event.Author.Username)
 		eb := embeds.NewBuilder()
 
+		// Builds embed title and description.
 		eb.Title(locale.Title)
 		eb.Description(locale.Description)
 
+		// Iterates and constructs crosspost groups into groupList.
+		var groupList [2][][2]string
```
Comment (VTGare) [inline commands/user.go:186]: Could you please explain to me what this array does, it's a bit confusing

Comment (XangelMusic) [reply commands/user.go:186]: First array is to seperate groups into Pairs and normal Groups.
Second, as a slice, is to get the full lists of both Pairs and Groups.
Third array is to store the embed name and description of each crosspost group.

**Comment by VTGare** [comment]
I made some changes to the code, but that's not all. We've talked about it on Discord but I'll mention it here: I don't think pairs should have a parent channel since both of them are parents, instead parent should be an empty string and both channels in the pair should be in the children array. Don't forget to sort children with `sort.Strings` https://pkg.go.dev/sort#Strings function before adding them to the database and also don't forget to sort them before comparing them to figure out whether such a pair already exists (the whole point of sorting).

Another thing, I wonder if it's better to use an enum instead of an isPair boolean to maybe make adding more support to group types in the future easier. In any case, if we're keeping the boolean, then it should be called `is_pair` in the database.

**Review by VTGare — APPROVED** [review]
LGTM.

**Pull request merged**